File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int ar = 0;
        int curar;
        int i=0;
        int j=height.length-1;
        while(i<j)
        {
            curar = (j-i)*(Math.min(height[i],height[j]));
            ar = Math.max(ar,curar);
            if (height[i]<height[j])
            {
                i++;
            }
            else
            {
                j--;
            }
        }
        return ar;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        area = 0
        currentarea = 0
        i = 0
        j = len(height) -1
        while(i<j):
            currentarea = (j-i)*min(height[i],height[j])
            area = max(area,currentarea)
            if(height[i]<height[j]):
                i +=1
            else:
                j -=1
        return area